### PR TITLE
Fix regexp for cleaning whitespaces

### PR DESCRIPTION
**Pull request by inercia**
## Description

* Try to improve the output by keeping indentation while removing some extra spaces. Maybe this we could remove even more spaces...
* Add the [`Masterminds/sprig`](https://github.com/Masterminds/sprig)  template funcs. Now we can do things like `{{ safe (renderComments .CommentLines) | fromHTML | indent 4 | toHTML }}`

## Example

With this code

```html
<tr>
    <td>
        <code>manifest</code></br>
        <em>

                string

        </em>
    </td>
    <td>

    </td>
</tr>

    </tbody>
</table>

<h3 id="getambassador.io/v2.ConditionStatus">ConditionStatus
    (<code>string</code> alias)</p></h3>

    <p>
        (<em>Appears on:</em>
            <a href="#getambassador.io/v2.AmbInsCondition">AmbInsCondition</a>)
    </p>

<p>

</p>

    <hr/>
```

Without this code

```html
<tr>
<td>
<code>appVersion</code></br>
<em>
string
</em>
</td>
<td>
</td>
</tr>
<tr>
<td>
<code>manifest</code></br>
<em>
string
</em>
</td>
<td>
</td>
</tr>
</tbody>
</table>
<h3 id="getambassador.io/v2.ConditionStatus">ConditionStatus
(<code>string</code> alias)</p></h3>
<p>
(<em>Appears on:</em>
<a href="#getambassador.io/v2.AmbInsCondition">AmbInsCondition</a>)
</p>
<p>
</p>
```

### CODE REVIEW

**Comment by ahmetb** [comment]
Sorry, I have noticed this repository just now. Must have gotten lost among my emails.

1. I'm not fully clear what this fixes? Can you give an example where the original code is doing a mistake? 
2. I'm not familiar with `sprig`, what do we get out of it here in particular?
3. Have you got this working for Ambassador?

**Comment by inercia** [comment]
Sorry, @ahmetb, I didn't see your reply before. The problem I was trying to fix was that, without this code, all the spaces are removed, and that means that I could not generate a valid Markdown (as indentation is important in Markdown). My example in the description of the PR is probably confusing (because it seems to be a cosmetic problem), but it is important when trying to generate Markdown. 

Regarding the use of `sprig`, we are using some of the functions provided there for generating our Markdown. We are currently using this for generating the auto-docs for the Ambassador Operator [here](https://github.com/datawire/ambassador-operator/blob/master/docs/api/index.md). You can see the Markdown templates [here](https://github.com/datawire/ambassador-operator/tree/master/docs/api/templates).

**Comment by ahmetb** [comment]
For many reasons markdown output won’t work. That’s why we generate html that has .md extension. Github and almost all md renderers render it correctly. I highly recommend sticking to html.

File: main.go [hunk 1]
```diff
@@ -146,7 +147,7 @@ func main() {
 		}
 
 		// remove trailing whitespace from each html line for markdown renderers
-		s := regexp.MustCompile(`(?m)^\s+`).ReplaceAllString(b.String(), "")
```
Comment (jpeach) [inline main.go:149]: This regex matches leading whitespace, not trailing whitespace. AFAICT, this defeats attempts to use `<pre>` tags in doc comments, which are super helpful for providing YAML examples of CRD usage. @ahmetb would you take a PR that just adjusts the regex?

Comment (ahmetb) [reply main.go:149]: It was my intention to remove the LEADING whitespace here. the // comment is incorrect.

Output of this program is a markdown file (with HTML syntax). For that to work on github properly, we have to unindent all HTML lines.

**Pull request closed without merging**